substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProvider.java

```diff
@@ -101,8 +101,12 @@
             this.javaType = declaringClass.getWrapped();
             this.compilation = compilation;
             SourceManager sourceManager = ImageSingletons.lookup(SourceManager.class);
-            fullFilePath = sourceManager.findAndCacheSource(javaType, clazz);
-            this.cachePath = sourceManager.getCachePathForSource(javaType);
+            try (DebugContext.Scope s = debugContext.scope("DebugCodeInfo", declaringClass)) {
+                fullFilePath = ImageSingletons.lookup(SourceManager.class).findAndCacheSource(javaType, clazz, debugContext);
```

Comment:
Why not
```suggestion
                fullFilePath = sourceManager.findAndCacheSource(javaType, clazz, debugContext);
```
?

Reply:
Thanks for spotting this, Foivos. I agree I should change this here and also introduce a local var for sourceManager at line 320 to avoid the repeated lookups.

Fix pushed.